PythonAPI/CMakeLists.txt

```diff
@@ -104,16 +89,136 @@ target_link_libraries (
   rpc
 )
 
+set (
+  PYTHON_API_SOURCES
+  ${CARLA_PYTHON_API_SOURCE_PATH}/PythonAPI.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Actor.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Blueprint.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Client.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Commands.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Control.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Exception.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Geometry.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/LightManager.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Map.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Sensor.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/SensorData.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Snapshot.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/TrafficManager.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/Weather.cpp
+  ${CARLA_PYTHON_API_SOURCE_PATH}/World.cpp
+)
+
+if (ENABLE_OSM2ODR)
+  list (
+    APPEND
+    PYTHON_API_SOURCES
+    ${CARLA_PYTHON_API_SOURCE_PATH}/OSM2ODR.cpp
+  )
+endif ()
+
+if (ENABLE_RSS)
+  list (
+    APPEND
+    PYTHON_API_SOURCES
+    ${CARLA_PYTHON_API_SOURCE_PATH}/AdRss.cpp
+  )
+endif ()
+
+set (
+  USE_LIMITED_API
+  OFF
+)
+
+set (
+  PYTHON_API_ADD_LIBRARY_ARGS
+)
+
+if (USE_LIMITED_API)
+  list (
+    APPEND
+    PYTHON_API_ADD_LIBRARY_ARGS
+    USE_SABI ${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}
+  )
+endif ()
+
+list (
+  APPEND
+  PYTHON_API_ADD_LIBRARY_ARGS
+  WITH_SOABI
+)
+
+Python3_add_library (
+  carla-python-api-extension
+  MODULE
+  ${PYTHON_API_ADD_LIBRARY_ARGS}
+  ${PYTHON_API_SOURCES}
+)
+
+if (BUILD_PYTHON_API_WHEEL_PACKAGE)
+
+  add_custom_command (
+    TARGET carla-python-api-extension POST_BUILD
+    COMMAND
+      ${CMAKE_COMMAND}
+      -E rm
+      -rf
+      ${CARLA_PYTHON_API_CARLA_PATH}/carla
+    COMMAND
+      ${CMAKE_COMMAND}
+        -E copy
+        ${CARLA_WORKSPACE_PATH}/LICENSE
+        ${CARLA_PYTHON_API_CARLA_PATH}/LICENSE
+    COMMAND
+      ${CMAKE_COMMAND}
+        -E copy
+        $<TARGET_FILE:carla-python-api-extension>
+        ${CARLA_PYTHON_API_CARLA_PATH}/carla/carla_ext${PYTHON_MODULE_EXT}
+    COMMAND
+      ${CARLA_PYTHON_API_BUILD_COMMAND}
+    WORKING_DIRECTORY
+      ${CARLA_PYTHON_API_CARLA_PATH}
+    VERBATIM
+    USES_TERMINAL
+  )
+
+  add_custom_target (
+    carla-python-api-install
+    COMMAND
```

Comment:
What would happen here if python api is already installed? Should we try it uninstall it first?